data/data/agent/systemd/units/pre-network-manager-config.service

```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Prepare network manager config content
 Before=dracut-initqueue.service
```

Comment:
I'm confused by this service (which is copied verbatim [from assisted](https://github.com/openshift/assisted-service/blob/master/internal/ignition/templates/pre-network-manager-config.service)). The Before/After here suggest that we're expecting it to run during the initramfs. But below we have `WantedBy=multi-user.target`, not `initrd.target`.
Not that making it wanted by `initrd.target` would work (I don't think that Ignition lets you add services to run in the initramfs). I'm not sure that anything short of a dracut module is going to make it actually work in the initramfs.
As far as I can see, this is going to run _after_ the pivot, and the Before/After on the dracut services will be ignored because they're not accompanied by Wants/Requires and those services are not enabled in the multi-user target.

Looking back at the history, the NetworkManager ordering used to exist, but was removed and replaced with the dracut stuff: https://github.com/openshift/assisted-service/pull/1870
This was based on [this recommendation](https://bugzilla.redhat.com/show_bug.cgi?id=1959961#c30), which in turn was predicated on the [assertion](https://bugzilla.redhat.com/show_bug.cgi?id=1959961#c29) that this service was already running during the initramfs. The reason given for doing that (so that the network config can be used to load the rootfs from the minimal ISO) make sense, and from day 1 the script included [code intended to handle](https://github.com/openshift/assisted-service/blob/a714f40bf720c30d8e491bdf54e847fb468c0654/internal/constants/scripts.go#L53-L54) this case, but when it was run in the initramfs it was done by writing the same script into a dracut hook, and then recently [changed](https://github.com/openshift/assisted-service/pull/5107) to a _separate_ systemd service. _This_ systemd service (the one used for the non-minimal ISO) has always been the one that runs after the pivot, and it should always have been constrained to run before NetworkManager.

I think we should clean up these spurious dracut references so that we don't confuse anybody else. And also make the same change in the assisted discovery ISO.

Reply:
Thanks for the detailed analysis @zaneb. I think this is what is needed in assisted service: https://github.com/openshift/assisted-service/pull/5375

Reply:
Thanks for tracking that down @zaneb ! Updated to remove the dracut references.